src/bsoncxx/types/value.cpp

```diff
@@ -84,8 +90,8 @@ value::~value() {
     }
 }
 
-bsoncxx::type value::type() const {
-    return _type;
+stdx::optional<bsoncxx::type> value::type() const {
+    return _type ? stdx::optional<bsoncxx::type>(static_cast<bsoncxx::type>(_type)) : stdx::nullopt;
```

Comment:
Isn't _type already a bsoncxx::type? I'm also still curious about whether this can be simplified. Does

```
return _type ? {_type} : {stdx::nullopt};
```

not work?